### PR TITLE
Feature/34 integrate passportjs

### PR DESCRIPTION
**Pull request by prashantpawar**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -16,16 +16,21 @@
...
     "babel-core": "^6.3.17",
+    "babel-preset-es2015": "^6.3.13",
+    "babel-preset-stage-2": "^6.3.13",
     "bluebird": "^3.0.6",
     "body-parser": "^1.14.1",
+    "connect-redis": "^3.0.1",
+    "cookie-parser": "^1.4.0",
     "express": "^4.13.3",
+    "express-session": "^1.12.1",
+    "faker": "^3.0.1",
```
Comment (jackofseattle) [inline package.json]: How is faker a prod dependency?

File: server/config.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+var domain_name = process.env.DOMAIN_NAME || "http://localhost:8001"
+module.exports = {
+  redis_port : process.env.REDIS_PORT,
+  redis_host : process.env.REDIS_HOST,
+  oauth2_client_id : process.env.OAUTH2_CLIENT_ID,
+  oauth2_client_secret : process.env.OAUTH2_CLIENT_SECRET,
+  oauth2_callback_url : domain_name + "/auth/google/callback"
```
Comment (jackofseattle) [inline server/config.js]: You do this here and then you also construct the callback url in your ensureAuthenticated middleware.  Which one is the one that's supposed to win?

File: package.json (hunk 1 above)
Comment (prashantpawar) [inline package.json]: Ha, because of /populate method

File: server/config.js (hunk 2 above)
Comment (prashantpawar) [inline server/config.js]: Good catch, the line in ensureAuthenticated has no side effects, forgot to remove it.

File: package.json (hunk 1 above)
Comment (jackofseattle) [inline package.json]: Lets make that method a dev only route then.
```javascript
if (!process.env.NODE_ENV === 'production') {
  // Add populate route
}
```

File: server/server.js [hunk 3]
```diff
@@ -21,17 +28,80 @@ if (process.env.NODE_ENV !== 'production') {
...
-  });
+app.use(cookieParser());
+
+app.use(session({
+  store: new RedisStore({
+    prefix: 'vivace.sess:',
+    port: config.redis_port,
+    host: config.redis_host
+  }),
+  secret: 'get smarter',
```
Comment (jackofseattle) [inline server/server.js]: Config a secret here for the second option.

Comment (jackofseattle) [inline server/server.js]: ```'get smarter' || process.env.SESSION_SECRET```

Comment (prashantpawar) [inline server/server.js]: Done

File: package.json (hunk 1 above)
Comment (prashantpawar) [inline package.json]: Done

**Comment by jackofseattle** [comment]
LGTM

**Pull request merged**